infor-design/enterprise-wc
# 1045 grid editing remaining

**Explain the details for making this change. What existing problem does the pull request solve?**
Added remaining features from ticket #1045 
* add multiple rows at specific index
* add support for datepicker editor
* add support for timepicker editor
* addNewAtEnd setting to auto create/append rows in editable mode

**Related github/jira issue (required)**:
Closes #1045 

**Steps necessary to review your pull request (required)**:
1. Checkout branch, `npm install`, `npm run start`
2. Go to http://localhost:4300/ids-data-grid/editable.html

**Test Datepicker Editor**
* Click on a publishDate cell
* Date value should be autoselected
* Try selecting dates from date picker popup
* Cell should have dirty indicator if value changed

**Test Timepicker Editor**
* Click on a publishTime cell
* Time value should be autoselected
* Try selected times from time picker popup
* Cell should have dirty indicator if value changed

**Test addRows()**
* Open dev  console and run ```$('ids-data-grid').addRows([{}, {}, {}], 2)```
* It should insert 3 empty rows at row index 2
* Try different variations of rowData/rowIndex values

**Test addNewAtEnd**
1. Click on an editable cell on the last row
2. Hit Tab until a new row is appended to the data grid
3. Editable cells should be activated on focus

**Included in this Pull Request**:
